rust/src/operations/update.rs

```diff
@@ -0,0 +1,947 @@
+
+        let (table, metrics) = DeltaOps(table)
+            .update()
+            .with_update("modified", lit("2023-05-14"))
+            .await
+            .unwrap();
+
+        assert_eq!(table.version(), 2);
+        assert_eq!(table.get_file_uris().count(), 1);
+        assert_eq!(metrics.num_added_files, 1);
+        assert_eq!(metrics.num_removed_files, 1);
+        assert_eq!(metrics.num_updated_rows, 4);
+        assert_eq!(metrics.num_copied_rows, 0);
+
+        let expected = vec![
+            "+----+-------+------------+",
+            "| id | value | modified   |",
+            "+----+-------+------------+",
+            "| A  | 1     | 2023-05-14 |",
+            "| A  | 10    | 2023-05-14 |",
+            "| A  | 100   | 2023-05-14 |",
+            "| B  | 10    | 2023-05-14 |",
+            "+----+-------+------------+",
+        ];
+        let actual = get_data(&table).await;
+        assert_batches_sorted_eq!(&expected, &actual);
+    }
+
+    #[tokio::test]
+    async fn test_update_non_partition() {
+        let schema = get_arrow_schema(&None);
+        let table = setup_table(None).await;
+
+        let batch = RecordBatch::try_new(
+            Arc::clone(&schema),
+            vec![
+                Arc::new(arrow::array::StringArray::from_slice(["A", "B", "A", "A"])),
+                Arc::new(arrow::array::Int32Array::from_slice([1, 10, 10, 100])),
+                Arc::new(arrow::array::StringArray::from_slice([
+                    "2021-02-02",
+                    "2021-02-02",
+                    "2021-02-03",
+                    "2021-02-03",
+                ])),
+            ],
+        )
+        .unwrap();
+
+        let table = write_batch(table, batch).await.unwrap();
+        assert_eq!(table.version(), 1);
+        assert_eq!(table.get_file_uris().count(), 1);
+
+        let (table, metrics) = DeltaOps(table)
+            .update()
+            .with_predicate(col("modified").eq(lit("2021-02-03")))
+            .with_update("modified", lit("2023-05-14"))
+            .await
+            .unwrap();
+
+        assert_eq!(table.version(), 2);
+        assert_eq!(table.get_file_uris().count(), 1);
+        assert_eq!(metrics.num_added_files, 1);
+        assert_eq!(metrics.num_removed_files, 1);
+        assert_eq!(metrics.num_updated_rows, 2);
+        assert_eq!(metrics.num_copied_rows, 2);
+
+        let expected = vec![
+            "+----+-------+------------+",
+            "| id | value | modified   |",
+            "+----+-------+------------+",
+            "| A  | 1     | 2021-02-02 |",
+            "| A  | 10    | 2023-05-14 |",
+            "| A  | 100   | 2023-05-14 |",
+            "| B  | 10    | 2021-02-02 |",
+            "+----+-------+------------+",
+        ];
+        let actual = get_data(&table).await;
+        assert_batches_sorted_eq!(&expected, &actual);
+    }
+
+    #[tokio::test]
+    async fn test_update_partitions() {
```

Comment:
Since a partition-only predicate is a separate code path from one with a mix of partition and normal columns, I think we are missing some coverage on partition column handling in the case where we need to scan in find files.

Could you add either a separate test or just another part in this test where you update a partitioned table with a predicate that is on the partition column *and* a normal column?

Reply:
Done